From ef4eede6c3cb0a1eddf761808f3a76c9370e9159 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Tue, 26 Nov 2013 14:09:35 +0800
Subject: [PATCH 17/30] ENGR00289630 EGL load software opengl on mx6sl_evk

load software opengl es on mx6sl_evk as no GPU 3D on the chip.
On KitKat, egl loader will load SW opengl or HW opengl, and
default it will skip load software opengl except on emulater,
here we let egl loader load software opengl for mx6sl, and
load hardware opengl for mx6dq.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 opengl/libs/Android.mk     | 3 +++
 opengl/libs/EGL/Loader.cpp | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/opengl/libs/Android.mk b/opengl/libs/Android.mk
index b4f9058..ec7cb3c 100644
--- a/opengl/libs/Android.mk
+++ b/opengl/libs/Android.mk
@@ -44,6 +44,9 @@ LOCAL_CFLAGS += -DLOG_TAG=\"libEGL\"
 LOCAL_CFLAGS += -DGL_GLEXT_PROTOTYPES -DEGL_EGLEXT_PROTOTYPES
 LOCAL_CFLAGS += -fvisibility=hidden
 LOCAL_CFLAGS += -DEGL_TRACE=1
+ifneq ($(HAVE_FSL_IMX_GPU3D),true)
+LOCAL_CFLAGS += -DUSE_SW_OPENGL
+endif
 
 ifeq ($(BOARD_ALLOW_EGL_HIBERNATION),true)
   LOCAL_CFLAGS += -DBOARD_ALLOW_EGL_HIBERNATION
diff --git a/opengl/libs/EGL/Loader.cpp b/opengl/libs/EGL/Loader.cpp
index 02914a0..05e39b4 100644
--- a/opengl/libs/EGL/Loader.cpp
+++ b/opengl/libs/EGL/Loader.cpp
@@ -333,8 +333,11 @@ void *Loader::load_driver(const char* kind,
                         continue;
                     }
                     if (!strcmp(e->d_name, "libGLES_android.so")) {
+                        // don't skip software render if define USE_SW_OPENGL
+#ifndef USE_SW_OPENGL
                         // always skip the software renderer
                         continue;
+#endif
                     }
                     if (strstr(e->d_name, pattern.string()) == e->d_name) {
                         if (!strcmp(e->d_name + strlen(e->d_name) - 3, ".so")) {
-- 
1.8.0

